src/systems/filecoin_mining/storage_mining/storage_mining_subsystem.go

```diff
@@ -288,12 +288,10 @@ func (sms *StorageMiningSubsystem_I) VerifyElectionPoSt(inds indices.Indices, he
 		Output_: onChainInfo.Randomness(),
 	}
 
-	// get worker key from minerAddr
-	workerKey, err := sma.Info().Worker().GetKey()
-	if err != nil {
-		return false
-	}
-
+	// TODO if the workerAddress is secp then payload will be the blake2b hash of its public key
+	// and we will need to recover the entire public key from worker before handing off to Verify
+	// example of recover code: https://github.com/ipsn/go-secp256k1/blob/master/secp256.go#L93
+	workerKey := sma.Info().Worker().Payload()
```

Comment:
This code (and the below bits with a TODO) will not be executed anytime soon, but flagging that we will need to account for different key types before executing it.